transport/http/peer_test.go

```diff
@@ -102,6 +104,43 @@ func TestHTTPWithRoundRobin(t *testing.T) {
 	integrationtest.Blast(ctx, t, c)
 }
 
+func TestHTTPOnSuspect(t *testing.T) {
+	server, serverAddr := spec.NewServer(t, ":0")
+
+	client, c := spec.NewClient(t, []string{serverAddr})
+	defer client.Stop()
+
+	// Exercise OnSuspect
+	ctx := context.Background()
+	ctx, cancel1 := context.WithTimeout(ctx, 5*testtime.Millisecond)
+	defer cancel1()
```

Comment:
do we need a new `cancel` name? should we just do `ctx, cancel = context.WithTimeout(...)`?